smoothchecker.py

```diff
@@ -186,60 +173,50 @@ def check_chunks(base_url, manifest, stream_index, quality_level, processes):
-    request.get_method = lambda: 'HEAD'
-    return urllib2.urlopen(request)
+        return chunk_url, http_client.head(chunk_url).status_code
+    except HTTPError as e:
+        print(f"{e.request.url} returned {e.response.status_code}")
+        return e.request.url, e.response.status_code
 
 
 def results_join(output_file):
     with open(output_file, 'wb') as outfile:
-        for result_file in glob.glob('results/*.csv'):
+        for result_file in glob('results/*.csv'):
             with open(result_file, 'rb') as readfile:
-                shutil.copyfileobj(readfile, outfile)
+                copyfileobj(readfile, outfile)
 
 
 def options_parser():
-    version = "%%prog %s" % __version
-    usage = "usage: %prog [options] <manifest URL or file>"
-    parser = OptionParser(usage=usage, version=version,
-                          description=__description, epilog=__author_info)
-    parser.add_option("-i", "--info",
-                      action="store_true", dest="info_only",
-                      default=False, help="print Manifest info and exit")
-    parser.add_option("-m", "--manifest-only",
-                      action="store_true", dest="manifest_only",
-                      default=False, help="download Manifest file and exit")
-    parser.add_option("-r", "--prepare-results",
-                      action="store_true", dest="results",
-                      default=False, help="")
-    parser.add_option("-d", "--dest-dir", metavar="<dir>",
-                      dest="dest_dir", default=tempfile.gettempdir(),
-                      help="destination directory")
-    parser.add_option("-p", "--parallel-processes", metavar="<int>",
-                      dest="processes", default=cpu_count() * 2,
-                      help="parallel processes to be launched")
-    return parser
-
-
-if __name__ == "__main__":
-
+    _parser = OptionParser(
+        usage="usage: %prog [options] <manifest URL or file>",
+        version=("%%prog %s" % __version),
+        epilog=__author_info)
+    _parser.add_option('-i', '--info',
+                       action='store_true', dest='info_only',
+                       default=False, help='print Manifest info and exit')
+    _parser.add_option('-m', '--manifest-only',
+                       action='store_true', dest='manifest_only',
+                       default=False, help='download Manifest file and exit')
+    _parser.add_option('-r', '--prepare-results',
+                       action='store_true', dest='results',
+                       default=False, help='')
+    _parser.add_option('-d', '--dest-dir', metavar='<dir>',
+                       dest='dest_dir', default=gettempdir(),
+                       help='destination directory')
+    _parser.add_option('-p', '--parallel-processes', metavar='<int>',
+                       dest='processes', default=cpu_count() * 2,
+                       help='parallel processes to be launched')
+    return _parser
+
+
```

Comment:
this double line is on purpose?

Reply:
yes

Reply:
@Bimblis for your reference https://www.python.org/dev/peps/pep-0008/#blank-lines